### PR TITLE
[ VariadicSplit ] Dynamic shape inference

### PR DESCRIPTION
**Pull request by jane-intel**

### CODE REVIEW

**Comment by raramer01** [comment]
TESTNOW_DLDT_E2E

**Comment by raramer01** [comment]
TESTNOW_DLDT_E2E

File: src/ngraph/op/variadic_split.cpp [hunk 1]
```diff
@@ -99,25 +99,31 @@ void ngraph::op::v1::VariadicSplit::validate_and_infer_types()
                     sum_of_splits += split_lengths[i];
                 }
             }
+            auto data_shape_dims = vector<Dimension>{data.get_partial_shape()};
+            auto dimension_at_axis = data_shape_dims.at(axis);
 
-            if (negative_one > 0)
+            if (negative_one > 0 && dimension_at_axis.is_static())
```
Comment (mitruska) [inline src/ngraph/op/variadic_split.cpp]: I just wonder, what if we get `-1` as a first element, so at the `[0]` index of the `split_lengths` vector
```suggestion
            if (negative_one >= 0 && dimension_at_axis.is_static())
```

Comment (postrational) [reply src/ngraph/op/variadic_split.cpp]: Well spotted!

Comment (postrational) [reply src/ngraph/op/variadic_split.cpp]: I added a test to cover this.

**Pull request merged**